pkg/alertmanager/operator.go

```diff
@@ -475,6 +726,281 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+func (c *Operator) provisionAlertmanagerConfiguration(ctx context.Context, am *monitoringv1.Alertmanager, store *assetStore) error {
+
+	secretName := defaultConfigSecretName(am.Name)
+	if am.Spec.ConfigSecret != "" {
+		secretName = am.Spec.ConfigSecret
+	}
+
+	secret, err := c.kclient.CoreV1().Secrets(am.Namespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "get base configuration secret")
+	}
+
+	var rawBaseConfig []byte
+	if secret != nil && secret.Data["alertmanager.yaml"] != nil {
+		rawBaseConfig = secret.Data["alertmanager.yaml"]
+	} else {
+		level.Warn(c.logger).Log("msg", "either base config secret not found, or alertmanager.yaml field not specified",
+			"secret", secretName, "alertmanager", am.Name, "namespace", am.Namespace)
+
+		fallbackConfig := "route:\n  receiver: 'null'\n" +
+			"receivers:\n- name: 'null'"
+		rawBaseConfig = []byte(fallbackConfig)
+	}
```

Comment:
I'm wondering if we should have `Info` rather than `Warn`?

```suggestion
	rawBaseConfig := []byte(`route:
	receiver: 'null'
receivers:
- name: 'null'`)
	if secret != nil && secret.Data["alertmanager.yaml"] != nil {
		rawBaseConfig = secret.Data["alertmanager.yaml"]
	} else {
		level.Info(c.logger).Log("msg", "either base config secret not found, or alertmanager.yaml field not specified",
			"secret", secretName, "alertmanager", am.Name, "namespace", am.Namespace)
	}
```